net-misc/ubridge: New package

garrymar opened the pull request:
uBridge is a simple application to create user-land bridges between
various technologies. Currently bridging between UDP tunnels, Ethernet
and TAP interfaces is supported. Packet capture is also supported.

Bug: https://bugs.gentoo.org/642070
Package-Manager: Portage-2.3.19, Repoman-2.3.6

titanofold commented on net-misc/ubridge/metadata.xml:
```diff
@@ -0,0 +1,9 @@
+	<email>g.djavadyan@gmail.com</email>
+	<name>Garri Djavadyan</name>
+</maintainer>
+<longdescription>uBridge is a simple application to create user-land bridges between various technologies. Currently bridging between UDP tunnels, Ethernet and TAP interfaces is supported. Packet capture is also supported.</longdescription>
```
Need the Proxy project listed as well.

titanofold commented on net-misc/ubridge/ubridge-0.9.13.ebuild:
```diff
@@ -0,0 +1,39 @@
+}
+
+pkg_postinst() {
+
```
Delete extraneous blank line.

doughdemon commented:
From looking at the upstream Makefile:
1. The package should also depend on dev-libs/iniparser (automagic dependency)
2. The upstream Makefile does not honor CFLAGS, nor LDFLAGS

Fixing the second point probably requires a patch to the Makefile.

garrymar commented:
>     The package should also depend on dev-libs/iniparser (automagic dependency)
>     The upstream Makefile does not honor CFLAGS, nor LDFLAGS

Thank you for comments, I'll try to fix it.

garrymar commented:
@doughdemon, I see that upstream Makefile overrides CFLAGS, but I can't get why it doesn't respect LDFLAGS. Please verify updated PR. Thanks.

SoapGentoo commented on net-misc/ubridge/ubridge-0.9.13.ebuild:
```diff
@@ -0,0 +1,53 @@
+}
+
+src_prepare() {
+	epatch "${FILESDIR}"/${P}-respect-flags.patch
```
we dont want epatch and eutils in EAPI 6 anymore. Put the patch in global `PATCHES` array and nix `src_prepare`

SoapGentoo commented on net-misc/ubridge/ubridge-0.9.13.ebuild:
```diff
@@ -0,0 +1,53 @@
+IUSE="+system-iniparser"
+
+RDEPEND="
+	net-libs/libpcap
```
aren't these libs linked?

SoapGentoo commented on net-misc/ubridge/ubridge-0.9.13.ebuild:
```diff
@@ -0,0 +1,53 @@
+	else
+		unset SYSTEM_INIPARSER
+	fi
+	default
```
remove `default` and rename this function to `src_configure`

SoapGentoo commented on net-misc/ubridge/ubridge-0.9.13.ebuild:
```diff
@@ -0,0 +1,53 @@
+
+EAPI=6
+
+inherit eutils fcaps user
```
`eutils` can go once you got rid of `epatch`

garrymar commented:
@titanofold, @SoapGentoo many thanks for review. I've fixed the issues. The dev-libs/iniparser dependency uses any value specification because I've tested ubridge against both dev-libs/iniparser-3.1-r1:0 and dev-libs/iniparser-4.0:4 (masked) without recompiling it. In both cases ubridge successfully parsed ini configuration and started as expected.

mgorny commented:
It seems that @Whissi has committed this package already but I have reverted it because of major QA violation. I'm going to wait for him to respond before taking this further.

Whissi commented:
> Adding packages to maintainer-needed is a huge QA violation.

Well, I added the package when I bumped gns3 (non-maintainer bump) to fix the repository.
I am not going to maintain this package.
Should I commit next time with me as maintainer and drop 24h later to maintainer needed with mail to dev ml to workaround?

mgorny commented:
Then your developer status will be revoked due to intentional and purposeful violation of policy aka 'dumping work on others'. This was considered as a honest mistake.

mgorny commented:
Guys, i'm sorry for my tone. It was unnecessary and inappropriate here.

@garrymar, if you're still interested in this package, the way is free for it to be readded based upon your work. Just please let me know and I'll proceed with review and testing.

garrymar commented:
>if you're still interested in this package, the way is free for it to be readded based upon your work. Just please let me know and I'll proceed with review and testing.

@mgorny, thank you for your time. Yes, I'm still ready to maintain the package. I've pushed bumped version.

mgorny commented on net-misc/ubridge/ubridge-0.9.14.ebuild:
```diff
@@ -0,0 +1,52 @@
+
+inherit fcaps user
+
+DESCRIPTION="Bridge for UDP tunnels, Ethernet, TAP and VMnet interfaces."
```
Nit: this is not a full sentence, so no `.` at the end.

mgorny commented on net-misc/ubridge/ubridge-0.9.14.ebuild:
```diff
@@ -0,0 +1,52 @@
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+system-iniparser"
```
Is there a reason to ever *not* use system iniparser?

mgorny commented on net-misc/ubridge/ubridge-0.9.14.ebuild:
```diff
@@ -0,0 +1,52 @@
+
+RDEPEND="
+	net-libs/libpcap
+	system-iniparser? ( dev-libs/iniparser:* )
```
This is a C library, so I seriously doubt `:*` is ever valid here. Have you tested it against slot `4`? If both slots work, it should be `:=` to ensure that it is rebuilt when iniparser is upgraded. Otherwise, it should probably be `:0=` to force slot 0 and rebuild on upgrades within the slot with ABI changes (if those ever happen).

mgorny commented on net-misc/ubridge/ubridge-0.9.14.ebuild:
```diff
@@ -0,0 +1,52 @@
+}
+
+src_install() {
+	insinto /usr/bin
```
Please use `exeinto`/`exeopts`/`doexe` for executables. I know that you override all the stuff anyway but if someone adds non-executable files in the future, it would be cleaner.

mgorny commented on net-misc/ubridge/ubridge-0.9.14.ebuild:
```diff
@@ -0,0 +1,52 @@
+	fcaps -g ubridge -m 4710 -M 0710 cap_net_raw,cap_net_admin \
+		"${EROOT}"/usr/bin/ubridge
+
+	ewarn "NOTE: To read packets from the network interfaces with ubridge as"
```
Why warn? I think this is a regular log-class message.

mgorny commented on net-misc/ubridge/files/ubridge-0.9.14-respect-flags.patch:
```diff
@@ -0,0 +1,20 @@
++++ b/Makefile	2018-03-16 09:16:09.588132065 +0500
+@@ -40,7 +40,7 @@
+ 
+ CC      ?=   gcc
```
You need to explicitly write `src_compile` and call it as `emake CC="$(tc-getCC)"` (from toolchain-funcs) to respect CC correctly.

mgorny commented on net-misc/ubridge/files/ubridge-0.9.14-respect-flags.patch:
```diff
@@ -0,0 +1,20 @@
+ CC      ?=   gcc
+ 
+-CFLAGS  =   -O3 -Wall
++CFLAGS  =   -Wall
```
This still overrides user's CFLAGS. I suppose you want `+=` here.

garrymar commented:
> @mgorny requested changes on this pull request.

>> In net-misc/ubridge/ubridge-0.9.14.ebuild:
>> +DESCRIPTION="Bridge for UDP tunnels, Ethernet, TAP and VMnet
>> interfaces."
> Nit: this is not a full sentence, so no . at the end.

Got it. Done. Thanks.


>> In net-misc/ubridge/ubridge-0.9.14.ebuild:
>> +IUSE="+system-iniparser"
> Is there a reason to ever not use system iniparser?

I accustomed to Gentoo's facilities to provide full control to end users. So, I tried to align the ebuild to the Gentoo's philosophy. Please correct me if I am wrong. Thanks.


> In net-misc/ubridge/ubridge-0.9.14.ebuild:
> +	system-iniparser? ( dev-libs/iniparser:* )
> This is a C library, so I seriously doubt :* is ever valid here. Have
> you tested it against slot 4? If both slots work, it should be := to
> ensure that it is rebuilt when iniparser is upgraded. Otherwise, it
> should probably be :0= to force slot 0 and rebuild on upgrades within
> the slot with ABI changes (if those ever happen).

I actually tested it against both slots without recompiling and saw no problems (I reported it earlier on 25th of December). Nevertheless I've replaced :* to := as a safety measure.


>> In net-misc/ubridge/ubridge-0.9.14.ebuild:
>> +src_install() {
>> +	insinto /usr/bin
> Please use exeinto/exeopts/doexe for executables. I know that you
> override all the stuff anyway but if someone adds non-executable
> files in the future, it would be cleaner.

Got it. Done. Thanks.


>> In net-misc/ubridge/ubridge-0.9.14.ebuild:
>> +	ewarn "NOTE: To read packets from the network interfaces
>> with ubridge as"
> Why warn? I think this is a regular log-class message.

Because it is critical step to get working utility. Actually, I have adopted the message from net-analyzer/wireshark package.


>> In net-misc/ubridge/files/ubridge-0.9.14-respect-flags.patch:
>> + CC      ?=   gcc
> You need to explicitly write src_compile and call it as emake
> CC="$(tc-getCC)" (from toolchain-funcs) to respect CC correctly.

Thank you. Done. I have not paid attention to the CC.


>> In net-misc/ubridge/files/ubridge-0.9.14-respect-flags.patch:
>> +-CFLAGS  =   -O3 -Wall
>> ++CFLAGS  =   -Wall
> This still overrides user's CFLAGS. I suppose you want += here.

Sure. I have introduced the typo when bumped ubridge version. Thanks.

garrymar commented:
> We only provide control over what we consider useful & safe. If there is no noticeable difference between the two variants, there is no reason to have a USE flag. And given that bundling is strongly discouraged, that's one more reason to force system library.

@mgorny, the point is clear for me now. I have pushed updated version. Thank you, Michał!

mgorny commented on net-misc/ubridge/files/ubridge-0.9.14-respect-flags.patch:
```diff
@@ -0,0 +1,20 @@
+--- a/Makefile	2018-03-16 09:15:10.528245830 +0500
```
I hope you're going to submit this upstream? ;-)

garrymar replied:
> I hope you're going to submit this upstream? ;-)

Yes, I'll do it. Thanks.

garrymar replied:
Upstream has already merged the patch.

https://github.com/GNS3/ubridge/pull/52
